hackathon/01_health_check.ipynb

```diff
@@ -0,0 +1,228 @@
+    "\n",
+    "This hackathon is based on the [open-source Turbo Template](https://github.com/teamdatatonic/vertex-pipelines-end-to-end-samples).\n",
+    "Through this notebook series you'll get hands-on with the template and Google Cloud.\n",
+    "The hackathon is structured into the following exercises:\n",
+    "\n",
+    "1. **[Health check](./01_health_check.ipynb) - this notebook**\n",
+    "1. [Run pipelines](./02_run_pipelines.ipynb)\n",
+    "1. [Promote model](./03_promote_model.ipynb)\n",
+    "1. [Challenge: Model monitoring](./04_monitoring_challenge.ipynb)\n",
+    "1. [Challenge: Real-time predictions](./05_realtime_challenge.ipynb)\n",
+    "\n",
+    "\n",
+    "## Introduction\n",
+    "\n",
+    "In this notebook you'll health check your environment in Google Cloud before carrying out the next exercises and challenges.\n",
+    "Get started in [Vertex AI Workbench](https://cloud.google.com/vertex-ai-notebooks) now!\n",
+    "\n",
+    "**Prerequisites:**\n",
+    "\n",
+    "- [x] Deployed `dev`, `staging`, `prod` projects in Google Cloud\n",
+    "- [x] [Pyenv](https://github.com/pyenv/pyenv#installation) for managing Python versions\n",
+    "- [x] [Google Cloud SDK (gcloud)](https://cloud.google.com/sdk/docs/quickstart)\n",
+    "- [x] Make\n",
+    "- [x] [Poetry](https://python-poetry.org)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "122952d9-fb78-4f5d-8d8b-1e46bbe973b7",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Install tools\n",
+    "\n",
+    "Let's go to the root of the hackathon repo:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f95880c5-511e-4e4f-98de-6fbf7521ca61",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%cd .."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7faf7fe5-d425-4e70-beca-b641ae31640c",
+   "metadata": {},
+   "source": [
+    "Do you have poetry installed and available in the `PATH`?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e26f7572-24e5-4002-9181-65791d5426d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! poetry --version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "78e55311-0123-4f85-b5a8-390bb33638c3",
+   "metadata": {},
+   "source": [
+    "If not, install `pyenv` and `poetry`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "000357d7-bf37-4d3c-bfaa-70ed164f79b6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! curl https://pyenv.run | bash"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6bf595eb-7194-4b89-b512-d7db2fb87f24",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! bash ./docs/notebooks/scripts/install_poetry.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a188dda9-f3fb-498e-8b98-e9b2c4c2c444",
+   "metadata": {},
+   "source": [
+    "**Note:** After installation, restart your kernel or Workbench instance.\n",
```

Comment:
Could be worth adding this code to make things easier:

**_# Automatically restart kernel after installs
import os

if not os.getenv("IS_TESTING"):
    import IPython

    app = IPython.Application.instance()
    app.kernel.do_shutdown(True)_**